### ISSUE TITLE
onModelChange callback not working when using FroalaEditorComponent inside react web component with shadow root 

### ISSUE THREAD

**Issue by amitUpscale** [froala/react-froala-wysiwyg#388]
I have a react web component with shadow root, when I open the froala editor inside it by initializing editor FroalaEditorComponent everything works except for onModelChange function , this function does not return in model. it is always undefined.

**Comment by os-lmo** [comment]
I am also facing the same issue.
I know that, using the `fullPage` prop, we can overcome this issue. But, in that case, some other features will not work properly like popover position

**Comment by Amit-Milli** [comment]
Any update on this? this is very crucial for us!! When I use fullPage boolean prop as true all the css that we have written already gets affected. how can we make froala editor work seamlessly inside a shadowDOM?

**Comment by amitUpscale** [comment]
Any update on this? This is hurting us real bad.

**Comment by amitUpscale** [comment]
> I am also facing the same issue. I know that, using the `fullPage` prop, we can overcome this issue. But, in that case, some other features will not work properly like popover position

Were you able to find a solution for this?

**Comment by loganvolkers** [comment]
It looks like Froala may not support being rendered in a Shadow Root. Here's a small self-contained minimal example that shows it not working as expected: https://jsfiddle.net/m2z4ys8a/1/

**Comment by ilyaskarim** [comment]
Support for Shadow DOM in the WYSIWYG editors is limited, and it may lead to unexpected behavior and styling issues, as you have experienced. While loading the required styles in your app is a step in the right direction, there are certain limitations when it comes to the proper functioning of the editor within the Shadow DOM environment.
For that reason, Froala strongly recommend enabling the iframe/fullPage option to avoid unwanted conflicts in the editor functionality when working inside a shadow DOM element. 
Froala users are welcome to submit requests regarding their issues with iframe/fullPage, and we will provide solutions for them: https://froala.com/wysiwyg-editor/contact/

**Comment by emilianionascu** [comment]
Same issue here as well. What is it that prevents Froala from working properly in a Shadow DOM?

**Comment by Amit-Milli** [comment]
Yes, had to resort to using `iframe: true;` option in config. it solved the issue. 
you also need to add following options in config.
```
iframeStyleFiles: [/* All css files of your main app as iframe does not allow parent styles to get applied to children  */]
iframeStyle: `
            body {
              min-height: ## /* Set minimum height */
              max-height: ##; /* Set maximum height */
              z-index: ##;
              -webkit-font-smoothing: antialiased;
              font-family: Arial, Helvetica, sans-serif; /* default editor font size */
              font-size: 13px; /* default editor font family */
              color: #222;
              padding: ##;
            }`,
      // enter: FroalaEditor.ENTER_DIV,
    };

```

This will solve your problem. Now the only issue I am facing is that I am not able to see the tooltip values when I hover on toolbar which is at the bottom of editor. Does anyone have solution on this?

**Comment by hectordvv** [comment]
Enabling full page on editor worked for me.
![image](https://github.com/user-attachments/assets/dec5f52d-ba1d-42dc-8dc8-300634826d59)

**Comment by alexeyr-ci2** [comment]
> Froala users are welcome to submit requests regarding their issues with iframe/fullPage, and we will provide solutions for them: https://froala.com/wysiwyg-editor/contact/

The issue is https://github.com/froala/react-froala-wysiwyg/issues/274, the whole reason I wanted to use Shadow DOM is to get style encapsulation without `iframe: true`.
